src/azul/service/async_manifest_service.py

```diff
@@ -139,7 +147,7 @@ def _get_manifest_status(self, execution_id: str, request_index: int) -> Union[i
                 return 1
             else:
                 output = json.loads(output)
-                return output['Location']
+                return Manifest(**output)
```

Comment:
Still leaky. 

```suggestion
                return Manifest.from_json(output)
```

is more symmetrical and future-proof. It would allow us to modify the JSON representation without having to touch every callsite.